libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/HttpClient.kt

```diff
@@ -0,0 +1,77 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import androidx.annotation.Keep
+import com.mapbox.navigator.HttpCode
+import com.mapbox.navigator.HttpInterface
+import com.mapbox.navigator.HttpResponse
+import com.mapbox.services.android.navigation.BuildConfig
+import java.io.ByteArrayOutputStream
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.logging.HttpLoggingInterceptor
+import okio.Okio
+import timber.log.Timber
+
+@Keep
+internal class HttpClient(
+    private val userAgent: String,
+    private val acceptGzipEncoding: Boolean = false,
+    private val clientBuilder: OkHttpClient.Builder = OkHttpClient.Builder()
+) : HttpInterface() {
+
+    companion object {
+        private const val ERROR_EMPTY_USER_AGENT = "Empty UserAgent is not allowed"
+        internal const val HEADER_USER_AGENT = "User-Agent"
```

Comment:
Minor thing: I'd rather have these `private` (so we don't leak them in the production code) and "duplicate" them in `HttpClientTest`.

Reply:
fixed